scapy/contrib/gtp_v2.py

```diff
@@ -456,10 +478,28 @@ class IE_ULI(gtp.IE_Base):
 }
 
 
+class IE_UCI(gtp.IE_Base):
+    name = "IE UCI"
+    fields_desc = [ByteEnumField("ietype", 145, IEType),
+                   FieldLenField("length", None, length_of="UCI",
```

Comment:
Does this work ?! `length_of` expects a field name.5)343 is no field names "UCI". same everywhere

Reply:
Yes, it does.
In fact, I've also added tests to verify this functionality:
```
= IE_UCI, dissection
h = "fe1d70fa717ceeeeeeeeeeee080045000127a4f500003c11e9aec0a8ee80c0a87f50084b23a301131aa1482001070000000001020f009100080021f3540000001602"
gtp = Ether(hex_bytes(h))
ie = gtp.IE_list[0]
ie.CSG_ID == 22 and ie.AccessMode == 0 and ie.LCSG == 1 and ie.CMI == 0 and ie.MCC == b'123' and ie.MNC == b'45'

= IE_UCI, basic instantiation
ie = IE_UCI(ietype='UCI', length=8, CR_flag=0, instance=0, MCC=b'123', MNC=b'45', SPARE=0, CSG_ID=22, AccessMode=0, LCSG=1, CMI=0)
ie.ietype == 145 and ie.CSG_ID == 22 and ie.AccessMode == 0 and ie.LCSG == 1 and ie.CMI == 0 and ie.MCC == b'123' and ie.MNC == b'45'
```

Reply:
You're not building it.
what about
```
ie = IE_UCI(ietype='UCI', length=8, CR_flag=0, instance=0, MCC=b'123', MNC=b'45', SPARE=0, CSG_ID=22, AccessMode=0, LCSG=1, CMI=0)
bytes(or)
```

Reply:
You mean like this?
```
Python 3.6.8 (default, Aug  7 2019, 17:28:10) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os, binascii
>>> import scapy
>>> from scapy.all import *
>>> from scapy.contrib.gtp_v2 import *
>>> ie = IE_UCI(ietype='UCI', length=8, CR_flag=0, instance=0, MCC=b'123', MNC=b'45', SPARE=0, CSG_ID=22, AccessMode=0, LCSG=1, CMI=0)
>>> ie
<IE_UCI  ietype=UCI length=8 CR_flag=0 instance=0 MCC='123' MNC='45' SPARE=0 CSG_ID=22 AccessMode=0 SPARE=0 LCSG=1 CMI=0 |>
>>> gtpv2_pkt = Ether() / IP(dst='10.15.15.61') / UDP(sport=2123, dport=2123) / GTPHeader(seq=1, P=0, T=0, gtp_type=32, teid=1, length=16) / GTPV2CreateSessionRequest(IE_list=[ie])
>>> gtpv2_pkt
<Ether  type=IPv4 |<IP  frag=0 proto=udp dst=10.15.15.61 |<UDP  sport=gtp_control dport=gtp_control |<GTPHeader  P=0 T=0 gtp_type=create_session_req length=16 seq=1 |<GTPV2CreateSessionRequest  IE_list=[<IE_UCI  ietype=UCI length=8 CR_flag=0 instance=0 MCC='123' MNC='45' SPARE=0 CSG_ID=22 AccessMode=0 SPARE=0 LCSG=1 CMI=0 |>] |>>>>>
>>> 
```
Its working...

Reply:
Building means converting a packet to bytes. To build it you need to call `bytes()` on it

Reply:
Oh I get it now.
Any suggestions, how I can fix these?
```
>>> gtpv2_pkt.bytes()
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 366, in __getattr__
    fld, v = self.getfield_and_val(attr)
TypeError: 'NoneType' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 366, in __getattr__
    fld, v = self.getfield_and_val(attr)
TypeError: 'NoneType' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 366, in __getattr__
    fld, v = self.getfield_and_val(attr)
TypeError: 'NoneType' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 366, in __getattr__
    fld, v = self.getfield_and_val(attr)
TypeError: 'NoneType' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 366, in __getattr__
    fld, v = self.getfield_and_val(attr)
TypeError: 'NoneType' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 368, in __getattr__
    return self.payload.__getattr__(attr)
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 368, in __getattr__
    return self.payload.__getattr__(attr)
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 368, in __getattr__
    return self.payload.__getattr__(attr)
  [Previous line repeated 2 more times]
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 366, in __getattr__
    fld, v = self.getfield_and_val(attr)
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 1585, in getfield_and_val
    raise AttributeError(attr)
AttributeError: bytes
```

Reply:
I meant `bytes(gtpv2_pkt)`

Reply:
```
>>> bytes(gtpv2_pkt)
b'\xff\xff\xff\xff\xff\xff\x00\x00\x00\x00\x00\x00\x08\x00E\x00\x000\x00\x01\x00\x00@\x11H%\n\x0f\x0f=\n\x0f\x0f=\x08K\x08K\x00\x1cVb@ \x00\x10\x00\x00\x01\x00\x91\x00\x08\x00!\xf3T\x00\x00\x00\x16\x02'
```

Reply:
Could you only try `bytes(IE_UCI())`. I expect the computation of the length to be broken

Reply:
you are right; looks like it is broken:
```
>>> bytes(IE_UCI())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 492, in __bytes__
    return self.build()
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 612, in build
    p = self.do_build()
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 594, in do_build
    pkt = self.self_build()
  File "/usr/local/lib/python3.6/site-packages/scapy/packet.py", line 575, in self_build
    p = f.addfield(self, p, val)
  File "/usr/local/lib/python3.6/site-packages/scapy/fields.py", line 150, in addfield
    return s + self.struct.pack(self.i2m(pkt, val))
  File "/usr/local/lib/python3.6/site-packages/scapy/fields.py", line 1465, in i2m
    fld, fval = pkt.getfield_and_val(self.length_of)
TypeError: 'NoneType' object is not iterable
```